uber/models.py

```diff
@@ -1182,14 +1182,14 @@ def last_first(cls):
     @property
     def watchlist_guess(self):
         try:
-            with sa.Session() as session:
+            with Session() as session:
                 return session.guess_attendee_watchentry(self)
```

Comment:
Didn't we used to return the ``.to_dict()`` of the results here?  I guess that must have changed when @kitsuta did a refactor on the PR that implemented this stuff, and I didn't notice.

Returning an actual SQLAlchemy model object (or list of SQLAlchemy model objects) in a way where they survive the session being closed will cause problems.  For example:

```python
>>> with Session() as session:
...     a = session.query(Attendee).first()
... 
>>> a.first_name
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vagrant/uber/sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/attributes.py", line 237, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/home/vagrant/uber/sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/attributes.py", line 573, in get
    value = state._load_expired(state, passive)
  File "/home/vagrant/uber/sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/state.py", line 480, in _load_expired
    self.manager.deferred_scalar_loader(self, toload)
  File "/home/vagrant/uber/sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/loading.py", line 608, in load_scalar_attributes
    (state_str(state)))
sqlalchemy.orm.exc.DetachedInstanceError: Instance <Attendee at 0x7f2165e29550> is not bound to a Session; attribute refresh operation cannot proceed
```

That's why it's important to either call ``.to_dict()`` on whatever we're returning OR to use a session object which we know doesn't get closed when our function returns.

I suspect that was your problem here rather than ``listify`` since ``.all()`` just returns a plain list object:

```python
>>> xs = session.query(Attendee).all()
>>> type(xs)
<class 'list'>
```

I've confirmed that this is an issue on my local dev instance by creating a watchlist entry for "Some Jerk":

```python
>>> a = Attendee(first_name='Some', last_name='Jerk')
>>> a.banned
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vagrant/uber/sideboard/sideboard/lib/sa/_crud.py", line 1506, in __repr__
    for name in attr_names)
  File "/home/vagrant/uber/sideboard/sideboard/lib/sa/_crud.py", line 1506, in <genexpr>
    for name in attr_names)
  File "/home/vagrant/uber/sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/attributes.py", line 237, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/home/vagrant/uber/sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/attributes.py", line 573, in get
    value = state._load_expired(state, passive)
  File "/home/vagrant/uber/sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/state.py", line 480, in _load_expired
    self.manager.deferred_scalar_loader(self, toload)
  File "/home/vagrant/uber/sideboard/env/lib/python3.4/site-packages/sqlalchemy/orm/loading.py", line 608, in load_scalar_attributes
    (state_str(state)))
sqlalchemy.orm.exc.DetachedInstanceError: Instance <WatchList at 0x7f264c82ab38> is not bound to a Session; attribute refresh operation cannot proceed
```